crates/ursa-proxy/src/core/handler.rs

```diff
@@ -22,6 +27,90 @@ use tracing::{error, info, warn};
 
 type Client = client::Client<HttpConnector, Body>;
 
+pub async fn init_server_app<C: Cache>(
+    server_config: ServerConfig,
+    cache: C,
+    client: Client,
+) -> Router {
+    let mut user_apps = Router::new();
+    for location in server_config.clone().location {
+        let route_path = format!(
+            "/{}/*path",
```

Comment:
could we match for `/.well-known/acme-challenge/` ? certbot will only request from there